inference-engine/src/preprocessing/arm_neon/ie_preprocess_gapi_kernels_neon.cpp

```diff
@@ -475,6 +477,289 @@ void calcRowLinear_8U(C4, std::array<std::array<uint8_t*, 4>, 4>& dst,
+            v_int16 val1_3 = v_reinterpret_as_s16(v_expand_high(val_3));
+
+            v_int16 t0 = v_mulhrs(v_sub_wrap(val0_0, val1_0), a10);
+            v_int16 t1 = v_mulhrs(v_sub_wrap(val0_1, val1_1), a32);
+            v_int16 t2 = v_mulhrs(v_sub_wrap(val0_2, val1_2), a54);
+            v_int16 t3 = v_mulhrs(v_sub_wrap(val0_3, val1_3), a76);
+
+            v_int16 r0 = v_add_wrap(val1_0, t0);
+            v_int16 r1 = v_add_wrap(val1_1, t1);
+            v_int16 r2 = v_add_wrap(val1_2, t2);
+            v_int16 r3 = v_add_wrap(val1_3, t3);
+
+            v_uint8 q0 = v_pack_u(r0, r1);
+            v_uint8 q1 = v_pack_u(r2, r3);
+
+            v_uint8 q2 = v_shuffle(q0, hmask);
+            v_uint8 q3 = v_shuffle(q1, hmask);
+
+            v_uint8 q4 = v_blend<0xCC /*0b11001100*/>(q2, v_shift_left<4>(q3));
+            v_uint8 q5 = v_blend<0xCC /*0b11001100*/>(v_shift_right<4>(q2), q3);
+
+            v_store_low(&dst[0][x],  q4);
+            v_store_high(&dst[1][x], q4);
+            v_store_low(&dst[2][x],  q5);
+            v_store_high(&dst[3][x], q5);
+        }
+
+        if (x < length) {
+            x = length - half_nlanes;
+            continue;
+        }
+        break;
+    }
+}
+
+// 8UC1 Resize (bi-linear)
+void calcRowLinear_8UC1(uint8_t* dst[],
+                        const uint8_t* src0[],
+                        const uint8_t* src1[],
+                        const short    alpha[],
+                        const short    clone[],  // 4 clones of alpha
+                        const short    mapsx[],
+                        const short    beta[],
+                            uint8_t    tmp[],
+                        const Size&    inSz,
+                        const Size&    outSz,
+                        const int      lpi) {
+    static_assert(v_uint8::nlanes == 16,
+                  "The wide of NEON vector is 128 bits, so one vector contains 16 uchars");
+
+    constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+    constexpr int half_nlanes = nlanes / 2;
+
+    bool xRatioEq = inSz.width == outSz.width;
+    bool yRatioEq = inSz.height == outSz.height;
+
+    if (!xRatioEq && !yRatioEq) {
+        GAPI_Assert(inSz.width >= half_nlanes);
+        GAPI_Assert(outSz.width >= half_nlanes);
+
+        if (4 == lpi) {
+            // vertical pass
+            v_int16 b0 = vx_setall_s16(beta[0]);
+            v_int16 b1 = vx_setall_s16(beta[1]);
+            v_int16 b2 = vx_setall_s16(beta[2]);
+            v_int16 b3 = vx_setall_s16(beta[3]);
+
+            uchar _mask_vertical[nlanes] = { 0, 8, 4, 12, 1, 9, 5, 13,
+                                            2, 10, 6, 14, 3, 11, 7, 15 };
+            v_uint8 vmask = vx_load(_mask_vertical);
+
+            uchar _mask_horizontal[nlanes] = { 0, 4, 8, 12, 2, 6, 10, 14,
+                                              1, 5, 9, 13, 3, 7, 11, 15 };
+            v_uint8 hmask = vx_load(_mask_horizontal);
```

Comment:
As the horizontal mask is setting up in the function `horizontal_4LPI`, it's not necessary to allocate it here too. And it's not used further, as I can see

Reply:
Done.